tests/nil_undefined_test.go

```diff
@@ -0,0 +1,72 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+func foo2(x interface{}) string {
+	s := fmt.Sprintln("x:", x)
+	s += fmt.Sprintln("x == nil:", x == nil)
+	s += fmt.Sprintln("x == js.Undefined:", x == js.Undefined)
+	return s
+}
+
+func TestNilUndefinedEquality(t *testing.T) {
+	if got, want := (interface{})(nil) == (interface{})(nil), true; got != want {
+		t.Errorf("\n(interface{})(nil) == (interface{})(nil): got %v, want %v", got, want)
+	}
+	if got, want := js.Undefined == js.Undefined, true; got != want {
+		t.Errorf("\njs.Undefined == js.Undefined: got %v, want %v", got, want)
+	}
+	var ui interface{} = js.Undefined
+	if got, want := ui == js.Undefined, true; got != want {
+		t.Errorf("\nui == js.Undefined: got %v, want %v", got, want)
+	}
+	var u = js.Undefined
+	if got, want := u == js.Undefined, true; got != want {
+		t.Errorf("\nu == js.Undefined: got %v, want %v", got, want)
+	}
+	if got, want := u == ui, true; got != want {
+		t.Errorf("\nu == ui: got %v, want %v", got, want)
+	}
+	if got, want := (interface{})(nil) == js.Undefined, false; got != want {
```

Comment:
In pure Go `js.Undefined` is just a `nil` stub. So this comparison is comparing the interface values `(nil, nil)` and `(*js.Object, nil)`, which is `false` (the tricky `nil` situation with interfaces...). In GopherJS on the other hand, `(interface{})(nil)` gives JS's `null` and `js.Undefined` gives `undefined`, which are also unequal.